JavaScript/JavaScriptSDK/snippet.js

```diff
@@ -78,4 +78,8 @@
 
 // global instance must be set in this order to mitigate issues in ie8 and lower
 window.appInsights = appInsights;
-appInsights.trackPageView();
+
+// if somebody calls the snippet twice, don't report page view again
+if (appInsights.queue && appInsights.queue.length === 0) {
```

Comment:
Just to give some context. I was trying to find a solution which doesn't introduce a global var and is as small as possible. This code will be executed before the full library is downloaded and the snippet is only initializing an empty queue. 
Of course, you can break it if you put `appInsights.queue.push('haha')` before this code, but we have to assume that nobody will change the snippet code and assume it will still work.